scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
-    if args.srf is None or srf_name == args.srf:
-        # TODO: save status as HF. refer to submit_hf
-        # Use HF nt for wct estimation
-        nt = get_hf_nt(params)
-        fd_count = len(shared.get_stations(params.FD_STATLIST))
-
-        est_core_hours, est_run_time = est.est_BB_chours_single(fd_count, nt, ncores)
-
-        # creates and extra variable so we keep the original estimated run time for other purpos
-        est_run_time_scaled = est_run_time
-        if hasattr(args, "retries") and int(args.retries) > 0:
-            # check if BB.bin is read-able = restart-able
-            try:
-                from qcore.timeseries import BBSeis
-
-                bin = BBSeis(simulation_structure.get_bb_bin_path(params.sim_dir))
-            except:
-                logger.debug("Retried count > 0 but BB.bin is not readable")
-            else:
-                est_run_time_scaled = est_run_time * (int(args.retries) + 1)
-
-        wct = set_wct(est_run_time_scaled, ncores, args.auto)
-        write_directory = (
-            args.write_directory if args.write_directory else params.sim_dir
-        )
-        underscored_srf = srf_name.replace("/", "__")
-
-        header_dict = {
-            "wallclock_limit": wct,
-            "job_name": "bb.{}".format(underscored_srf),
-            "job_description": "BB calculation",
-            "additional_lines": "###SBATCH -C avx",
-            "platform_specific_args": get_platform_node_requirements(ncores),
-        }
-
-        body_template_params = (
-            "{}.sl.template".format(sl_name_prefix),
-            {"test_bb_script": "test_bb.sh"},
-        )
+    srf_name = Path(params.srf_file).stem
+
+    # TODO: save status as HF. refer to submit_hf
+    # Use HF nt for wct estimation
+    nt = get_hf_nt(params)
+    fd_count = len(shared.get_stations(params.FD_STATLIST))
+
+    est_core_hours, est_run_time = est.est_BB_chours_single(fd_count, nt, ncores)
+
+    # creates and extra variable so we keep the original estimated run time for other purpos
+    est_run_time_scaled = est_run_time
+    if retries > 0:
+        # check if BB.bin is read-able = restart-able
+        try:
+            from qcore.timeseries import BBSeis
+
+            bin = BBSeis(simulation_structure.get_bb_bin_path(params.sim_dir))
+        except:
+            logger.debug("Retried count > 0 but BB.bin is not readable")
+        else:
+            est_run_time_scaled = est_run_time * (retries + 1)
+
+    wct = set_wct(est_run_time_scaled, ncores, auto)
+    write_directory = write_directory if write_directory else params.sim_dir
+    if write_directory is None:
+        write_directory = params.sim_dir
+
+    underscored_srf = srf_name.replace("/", "__")
+
+    header_dict = {
+        "wallclock_limit": wct,
+        "job_name": f"bb.{underscored_srf}",
+        "job_description": "BB calculation",
+        "additional_lines": "###SBATCH -C avx",
```

Comment:
Given it's got three # it's not used by slurm, do we need this anymore? I'm not sure I've seen avx used since I started

Reply:
Good point.